### PR TITLE
Whitelist

### PR DESCRIPTION
**Pull request by jkuruzovich**
Whitelist is necessary if using GitHub authentication in the wild.
This will prevent all users and only enable authorized users.

### CODE REVIEW

File: images/hub/jupyterhub_config.py [hunk 1]
```diff
@@ -180,8 +180,12 @@ def generate_user_name(spawner):
 # Enable admins to access user servers
 c.JupyterHub.admin_access = get_config('admin.access')
 
+
 c.Authenticator.admin_users = get_config('admin.users', [])
 
+if get_config('whitelist.enabled', False):
```
Comment (yuvipanda) [inline images/hub/jupyterhub_config.py]: Is this explicit check needed?

**Comment by yuvipanda** [comment]
Thanks for the patch!

Two things:

1. Can you move the 'whitelist' option inside 'hub'? Keeping 'admin' outside was a mistake we need to fix, and I'd rather not add more top level entries
2. Can you add a 'null' default in values.yaml?

Thank you for the patch!

**Comment by ILoveDataSCIENCE8** [comment]
Thanks for the feedback!  I think I got the changes you suggested.  Just let me know if more changes needed.
Jason

**Comment by jkuruzovich** [comment]
Thanks for the feedback! I think I got the changes you suggested. Just let me know if more changes needed.
Jason

**Comment by jkuruzovich** [comment]
Thanks!

**Comment by choldgraf** [comment]
pinging @arokem as we ran into this problem with neurohackweek I believe...just making sure he sees it

**Pull request merged**